### PR TITLE
8335967: "text-decoration: none" does not work with "A" HTML tags

### PR DESCRIPTION
**Pull request by aivanov-jdk**
After I integrated [JDK-8326734](https://bugs.openjdk.org/browse/JDK-8326734), there's no way to remove text decoration from an HTML element. The most common HTML element is `<a>` which is usually rendered _underlined_. You could add an inline style or apply a rule with `text-decoration: none` to remove underline.

This fix restores this functionality.

The test contains five cases. The first case ensures the `<a>` element is rendered with underline by default. The second and third cases verify that underline is removed if `text-decoration: none` is added to the `style` attribute or is applied by a style-sheet rule.

The fourth and fifth cases verify that underline is preserved the containing element, `<p>` in this case, has `text-decoration: underline`.

These five cases render the same way in browsers. However, the fourth and fifth cases render without underline in JDKs without JDK-8326734.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8335967](https://bugs.openjdk.org/browse/JDK-8335967): "text-decoration: none" does not work with "A" HTML tags (**Bug** - P2)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)
 * [Dmitry Markov](https://openjdk.org/census#dmarkov) (@dmarkov20 - **Reviewer**)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20313/head:pull/20313` \
`$ git checkout pull/20313`

Update a local copy of the PR: \
`$ git checkout pull/20313` \
`$ git pull https://git.openjdk.org/jdk.git pull/20313/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20313`

View PR using the GUI difftool: \
`$ git pr show -t 20313`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20313.diff">https://git.openjdk.org/jdk/pull/20313.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20313#issuecomment-2248524609)

### CODE REVIEW

File: test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+import javax.swing.JEditorPane;
+import javax.swing.text.View;
+import javax.swing.text.html.CSS;
+
+/*
+ * @test
+ * @bug 8323801 8326734
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java:37]: Is it correct to have these bug ids in jtreg tag?
I think you should add 8335967 bug id .

File: test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ * @summary Tests different combination of 'underline' and 'line-through';
+ *          the text should render with both 'underline' and 'line-through'.
+ * @run main HTMLTextDecoration
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java:40]: I guess It is supposed to run `HTMLTextDecorationNone` test not `HTMLTextDecoration`
```suggestion
 * @run main HTMLTextDecorationNone
```

File: test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+import javax.imageio.ImageIO;
+import javax.swing.JEditorPane;
+import javax.swing.text.View;
+import javax.swing.text.html.CSS;
+
+/*
+ * @test
+ * @bug 8323801 8326734
+ * @summary Tests different combination of 'underline' and 'line-through';
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java:38]: Need to update summary ?

File: test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java (hunk 2 above)
Comment (aivanov-jdk) [reply test/jdk/javax/swing/text/html/HTMLDocument/HTMLTextDecorationNone.java]: Yes. I copied the test and forgot to update its metadata.

**Review by kumarabhi006 — APPROVED** [review]
Verified the test with the fix. It works as expected.

**Review by dmarkov20 — APPROVED** [review]

**Comment by honkar-jdk** [comment]
`text-decoration: none` works correctly for anchor tag. 
The output generated by the test matches with HTML editors.

**Review by honkar-jdk — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**